SNAPSHOT.md

```diff
@@ -0,0 +1,42 @@
+# GHW snapshots
+
+For GHW, snapshots are partial clones of the `/proc`, `/sys` (et. al.) subtrees taken into arbitrary
+machines, which GHW can consume later. "partial" is because the snapshot doesn't need to contain a
+complete copy of all the filesystem subtree (that is doable but inpractical). It only needs to contain
+the paths GHW cares about. The snapshot concept was introduced [to make GHW easier to test](https://github.com/jaypipes/ghw/issues/66).
+
+## Create and consume snapshot
+
+The recommended way to create snapshots for GHW is to use the `ghw-snapshot` tool.
+This tool is maintained by the GHW authors, and snapshot created with this tool are guaranteed to work.
+
+To consume the GHW snapshots, please check the `README.md` document.
+
+## Snapshot design and definitions
+
+The reminder of this document will describe how a snapshot looks like and provides rationale for all the major design decisions.
+Even though this document aims to provide all the necessary information to understand how GHW creates snapshots and what you should
+expect, we recommend to check also the [project issues](https://github.com/jaypipes/ghw/issues) and the `git` history to have the full picture.
+
+### Scope
+
+GHW support snapshot only on linux platform. This restriction may be lifted in future releases.
+Snapshots must be consumable in the following supported ways:
+1. (way 1) from docker (or podman), mounting them as volumes. See `hack/run-against-snapshot.sh`
+2. (way 2) using the environment variables `GHW_SNAPSHOT_*`. See `README.md` for the full documentation.
+
+Other combinations are possible, but are unsupported and may stop working any time.
+You should depend only on the supported ways to consume snapshots.
+
+### Snapshot content constraints
+
+Stemming from the use cases, the snapshot content must have the following properties:
+0. (constraint 0) MUST contain the same information as live system (obviously). Whatever you learn from a live system, you MUST be able to learn from a snapshot.
+1. (constraint 1) MUST NOT require any post processing before it is consumable besides, obviously, unpacking the `.tar.gz` on the right directory - and pointing GHW to that directory.
+2. (constraint 2) MUST NOT require any special handling special code path in GHW. From GHW perspective running against a live system or against a snapshot should be completely transparent.
+3. (constraint 3) MUST contain only data - no executable code is allowed ever. This makes snapshots trivially safe to share and consume.
```

Comment:
Let's add an additional constraint please:

4. MUST NOT contain any personally-identifiable data. Data gathered into a snapshot is for testing and troubleshooting purposes and should be safe to send to troubleshooters to analyze.

Reply:
that's a great idea! Added. I wonder if there is a way to make sure we uphold this constraint in new code besides careful review.